.github/workflows/test-go.yml

```diff
@@ -131,19 +137,33 @@ jobs:
         id: build
         run: |
             set -exo pipefail
-            export BINARY_TESTS="${{ steps.list-binary-tests.outputs.list }}"
-            (
-              echo -n "matrix="
-              jq -c --arg BINARY "${BINARY_TESTS}" \
+            matrix_file="matrix.json"
+            if [ "${{ inputs.binary-tests}}" == "true" ]; then
+              export BINARY_TESTS="${{ steps.list-binary-tests.outputs.list }}"
+              jq --arg BINARY "${BINARY_TESTS}" --arg BINARY_INDEX "${{ inputs.total-runners }}" \
                 '.include += [{
-                  "id": 16,
+                  "id": $BINARY_INDEX,
                   "estimatedRuntime": "N/A",
                   "packages": $BINARY,
-                  "description": "partition 16 - binary test packages"
-              }]' matrix.json
+                  "description": "partition $BINARY_INDEX - binary test packages"
+              }]' matrix.json > new-matrix.json
+              matrix_file="new-matrix.json" 
+            fi
+            # convert the json to a map keyed by id
```

Comment:
there's probably not value in the descriptions, but we do need something with the list of packages, and something else with the ids. I can simplify it so that everything else gets ignored